src/components/Messages/Message.tsx

```diff
@@ -11,14 +14,17 @@ import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 
 const Message: FC = () => {
-  const router = useRouter();
-  const address = router.query.address as string;
-  const messageState = useMessageStore((state) => state);
-  const { messages, conversations } = messageState;
+  const { push, query } = useRouter();
+  const address = query.address as string;
   const currentProfile = useAppStore((state) => state.currentProfile);
+  const messageState = useMessageStore((state) => state);
```

Comment:
I'm not a Zustand expert, so correct me if I'm wrong, but won't returning the complete state cause the component to rerender any time anything in the state changes?

It seems like we want to only select the items we care about, and use `zustand/shallow` when selecting multiple items.

https://docs.pmnd.rs/zustand/recipes/recipes

Reply:
Yes it's always good to select each items individually 🙇🏼 

Example: https://github.com/lensterxyz/lenster/blob/main/src/components/Layout.tsx#L35-L41